web/controllers/api/download_controller.ex

```diff
@@ -3,34 +3,42 @@ defmodule Castle.API.DownloadController do
 
   @redis Application.get_env(:castle, :redis)
   @bigquery Application.get_env(:castle, :bigquery)
+  @group_ttl 900
 
   plug Castle.Plugs.ParseInt, "podcast_id"
 
-  def index(conn, %{"podcast_id" => podcast_id}) do
-    %{assigns: %{time_from: from, time_to: to, interval: interval}} = conn
-
-    {data, meta} = @redis.interval "downloads.podcast.#{podcast_id}", from, to, interval, fn(new_from) ->
-      @bigquery.podcast_downloads(podcast_id, new_from, to, interval)
+  def index(%{assigns: %{interval: intv, group: group}} = conn, %{"podcast_id" => id}) do
+    {data, meta} = @redis.cached key("podcast.#{id}", group), @group_ttl, fn() ->
```

Comment:
After all the trouble I went through refactoring intervals, I can't use that `@redis.interval` cache here anyways.

Say I'm getting a 4 hour interval 15-minutes at a time.  The top 10 countries over the entire interval may not be the same 10 countries if I calculated them in a single 15-minutes.  So there's no way to partially cache this query - it's all or nothing!